akka-cluster-typed/src/test/java/akka/cluster/ddata/typed/javadsl/ReplicatorTest.java

```diff
@@ -110,50 +113,80 @@
       replicator.tell(new Replicator.Subscribe<>(Key, changedAdapter));
     }
 
+    public static Behavior<ClientCommand> create() {
+      return Behaviors.setup((ctx) -> {
+        // The distributed data types still need the implicit untyped Cluster.
+        // We will look into another solution for that.
+        Cluster node = Cluster.get(Adapter.toUntyped(ctx.getSystem()));
+        ActorRef<Replicator.Command> replicator = DistributedData.get(ctx.getSystem()).replicator();
+
+        return new Counter(replicator, node, ctx);
+      });
+    }
+
+    // #sample
+    // omitted from sample, needed for tests, factory above is for the docs sample
     public static Behavior<ClientCommand> create(ActorRef<Command> replicator, Cluster node) {
-      return Behaviors.setup(ctx -> new Client(replicator, node, ctx));
+      return Behaviors.setup(ctx -> new Counter(replicator, node, ctx));
     }
+    // #sample
 
     @Override
     public Behaviors.Receive<ClientCommand> createReceive() {
       return receiveBuilder()
-        .onMessage(Increment.class, cmd -> {
-          replicator.tell(
-            new Replicator.Update<>(Key, GCounter.empty(), Replicator.writeLocal(), updateResponseAdapter,
-              curr -> curr.increment(node, 1)));
-          return this;
-        })
-        .onMessage(InternalUpdateResponse.class, msg -> {
-          return this;
-        })
-        .onMessage(GetValue.class, cmd -> {
-          replicator.tell(
-            new Replicator.Get<>(Key, Replicator.readLocal(), getResponseAdapter, Optional.of(cmd.replyTo)));
-          return this;
-        })
-        .onMessage(GetCachedValue.class, cmd -> {
-          cmd.replyTo.tell(cachedValue);
-          return this;
-        })
-        .onMessage(InternalGetResponse.class, msg -> {
-          if (msg.rsp instanceof Replicator.GetSuccess) {
-            int value = ((Replicator.GetSuccess<?>) msg.rsp).get(Key).getValue().intValue();
-            ActorRef<Integer> replyTo = (ActorRef<Integer>) msg.rsp.request().get();
-            replyTo.tell(value);
-          } else {
-            // not dealing with failures
-          }
-          return this;
-        })
-        .onMessage(InternalChanged.class, msg -> {
-          GCounter counter = (GCounter) msg.chg.get(Key);
-          cachedValue = counter.getValue().intValue();
-          return this;
-        })
+        .onMessage(Increment.class, this::onIncrement)
+        .onMessage(InternalUpdateResponse.class, msg -> Behaviors.same())
+        .onMessage(GetValue.class, this::onGetValue)
+        .onMessage(GetCachedValue.class, this::onGetCachedValue)
+        .onMessage(InternalGetResponse.class, this::onInternalGetResponse)
+        .onMessage(InternalChanged.class, this::onInternalChanged)
```

Comment:
yes, this is the style we should use in all doc samples

method references or single line lambda